modules/rds/main.tf

```diff
@@ -0,0 +1,105 @@
+variable "subnet_ids" {
+  type        = list(string)
+  description = "Subnets (should be private) to host RDS instances."
+}
+
+variable "name_prefix" {
+  type = string
+}
+
+variable "db_storage_size" {
+  type        = number
+  description = "The allocated storage in gibibytes."
+}
+
+variable "db_storage_type" {
+  type        = string
+  description = "One of \"standard\" (magnetic), \"gp2\" (general purpose SSD), or \"io1\" (provisioned IOPS SSD)."
+}
+
+variable "db_engine" {
+  type        = string
+  description = "The database engine to use."
+}
+
+variable "db_instance_type" {
+  type        = string
+  description = "The instance type of the RDS instance."
+}
+
+variable "security_group_id" {
+  type        = string
+  description = "The security group grants the access to database."
+}
+
+variable "db_name" {
+  type = string
+}
+
+variable "db_username" {
+  type = string
+}
+
+variable "db_password" {
+  type = string
+}
+
+variable "tags" {
+  type    = map(string)
+  default = {}
+}
+
+variable "engine_version" {
+  type = string
+}
+
+resource "aws_iam_role" "rds_enhanced_monitoring" {
+  name_prefix        = var.name_prefix
+  assume_role_policy = data.aws_iam_policy_document.rds_enhanced_monitoring.json
+}
+
+resource "aws_iam_role_policy_attachment" "rds_enhanced_monitoring" {
+  role       = aws_iam_role.rds_enhanced_monitoring.name
+  policy_arn = "arn:aws:iam::aws:policy/service-role/AmazonRDSEnhancedMonitoringRole"
+}
+
+data "aws_iam_policy_document" "rds_enhanced_monitoring" {
+  statement {
+    actions = ["sts:AssumeRole"]
+    principals {
+      type        = "Service"
+      identifiers = ["monitoring.rds.amazonaws.com"]
+    }
+  }
+}
+
+resource "aws_db_subnet_group" "rds_private_subnet" {
+  subnet_ids = var.subnet_ids
+}
+
+resource "aws_db_instance" "default" {
+  identifier_prefix       = var.name_prefix
+  allocated_storage       = var.db_storage_size
+  storage_type            = var.db_storage_type
+  engine                  = var.db_engine
+  engine_version          = var.engine_version
+  instance_class          = var.db_instance_type
+  db_subnet_group_name    = aws_db_subnet_group.rds_private_subnet.name
+  multi_az                = true
+  backup_retention_period = 7
+  monitoring_interval     = 30
```

Comment:
These params here should be defined as a `variable` with a default value (or locals if there's a reason to not expose the knob, but I believe we want to expose those knobs).

Reply:
Done.